packages/app-extensions/src/actions/modules/actionHandlers/simpleAction.js

```diff
@@ -46,6 +47,18 @@ export function* invokeRequest(definition, selection, parent, params) {
         notifier.info('error', 'client.component.actions.validationError', validationErrorCompact(response.body.errors),
           'exclamation')
       )
+    } else if (response.body && response.body.params.downloadUrl) {
+      const fileResponse = yield call(rest.requestBytesSaga, response.body.params.downloadUrl, {
+        method: 'POST',
+        body: {
+          entity: selection.entityName,
+          selection,
+          parent,
+          ...params,
+          formProperties: definition.properties
+        }
+      })
+      yield call(download.downloadReadableStream, fileResponse.body, response.body.params.filename)
     } else {
       const success = response.body.success === true
```

Comment:
No success message needed here? or do you think a file download is sufficient?

Reply:
In my opinion the download is enough, a message that the file has started downloading doesn't seem to add any value. But I don't mind either way.

Of course, assuming this gets ported to a more general way of downloading files from actions with the whole websocket / notification business this will be obsolete anyway.